packages/react-aria-components/src/DropZone.tsx

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright 2022 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import {ButtonContext} from './Button';
+import {ContextValue, Provider, RenderProps, SlotProps, useContextProps, useRenderProps} from './utils';
+import {DropOptions, mergeProps, useButton, useClipboard, useDrop, useFocusRing, useHover, usePress, VisuallyHidden} from 'react-aria';
+import {InputContext} from './Input';
+import {LinkContext} from './Link';
+import React, {createContext, ForwardedRef, forwardRef, useRef} from 'react';
+export interface DropZoneRenderProps {
+  /**
+   * Whether the dropzone is currently hovered with a mouse.
+   * @selector [data-hovered]
+   */
+  isHovered: boolean,
+  /**
+   * Whether the dropzone is focused, either via a mouse or keyboard.
+   * @selector :focus
+   */
+  isFocused: boolean,
+  /**
+   * Whether the dropzone is keyboard focused.
+   * @selector [data-focus-visible]
+   */
+  isFocusVisible: boolean,
+  /**
+   * Whether the dropzone is the drop target.
+   * @selector [data-drop-target]
+   */
+  isDropTarget: boolean
+}
+
+export interface DropZoneProps extends Omit<DropOptions, 'getDropOperationForPoint' | 'onDropActivate' | 'onInsert' | 'onRootDrop' | 'onItemDrop' | 'onReorder'>, RenderProps<DropZoneRenderProps>, SlotProps {}
+
+export const DropZoneContext = createContext<ContextValue<DropZoneProps, HTMLDivElement>>(null);
+
+function DropZone(props: DropZoneProps, ref: ForwardedRef<HTMLDivElement>) {
+  [props, ref] = useContextProps(props, ref, DropZoneContext);
+  let {dropProps, isDropTarget} = useDrop({...props, ref});
+  let {hoverProps, isHovered} = useHover({});
+  let {focusProps, isFocused, isFocusVisible} = useFocusRing();
```

Comment:
the focus props for this are going to the wrong element I think, they should be on the hidden button

otherwise, if this should show a focus ring around the entire set of controls regardless, then you can specify `{within: true}` on the props for `useFocusRing`

but I think the focus props should be hooked up to the hidden button